src/wp-includes/rest-api/endpoints/class-wp-rest-site-health-controller.php

```diff
@@ -0,0 +1,244 @@
+<?php
+/**
+ * REST API: WP_REST_Site_Health_Controller class
+ *
+ * @package WordPress
+ * @subpackage REST_API
+ * @since 5.5.0
+ */
+
+/**
+ * Core class for interacting with Site Health elements.
+ *
+ * @since 5.5.0
+ *
+ * @see WP_REST_Controller
+ */
+class WP_REST_Site_Health_Controller extends WP_REST_Controller {
+
+	private $site_health;
+
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		if ( ! class_exists( 'WP_Site_Health' ) ) {
+			require_once ABSPATH . 'wp-admin/includes/class-wp-site-health.php';
+		}
+
+		$this->namespace = 'wp-site-health/v1';
+
+		$this->site_health = WP_Site_Health::get_instance();
+	}
+
+	/**
+	 * Register API routes.
+	 *
+	 * @since 5.5.0
+	 *
+	 * @see register_rest_route()
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			sprintf(
+				'/%s/%s',
+				'test',
+				'background-updates'
+			),
+			array(
+				'methods'             => 'POST',
+				'callback'            => array( $this, 'test_background_updates' ),
+				'permission_callback' => function() {
+					return $this->validate_request_permission( 'background_updates' );
+				},
+			)
+		);
+
+		register_rest_route(
+			$this->namespace,
+			sprintf(
+				'/%s/%s',
+				'test',
+				'loopback-requests'
+			),
+			array(
+				'methods'             => 'POST',
+				'callback'            => array( $this, 'test_loopback_requests' ),
+				'permission_callback' => function() {
+					return $this->validate_request_permission( 'loopback_requests' );
+				},
+			)
+		);
+
+		register_rest_route(
+			$this->namespace,
+			sprintf(
+				'/%s/%s',
+				'test',
+				'dotorg-communication'
+			),
+			array(
+				'methods'             => 'POST',
+				'callback'            => array( $this, 'test_dotorg_communication' ),
+				'permission_callback' => function() {
+					return $this->validate_request_permission( 'dotorg_communication' );
+				},
+			)
+		);
+
+		register_rest_route(
+			$this->namespace,
+			sprintf(
+				'/%s/%s',
+				'test',
+				'debug_enabled'
+			),
+			array(
+				'methods'             => 'POST',
+				'callback'            => array( $this, 'test_is_in_debug_mode' ),
+				'permission_callback' => function() {
+					return $this->validate_request_permission( 'debug_enabled' );
+				},
+			)
+		);
+
+		register_rest_route(
+			$this->namespace,
+			sprintf(
+				'/%s/%s',
+				'get',
+				'directory-sizes'
+			),
+			array(
+				'methods'             => 'POST',
+				'callback'            => array( $this, 'get_directory_sizes' ),
+				'permission_callback' => function() {
+					return $this->validate_request_permission( 'debug_enabled' ) && ! is_multisite();
+				},
+			)
+		);
+	}
+
+	/**
+	 * Validate if the current user can request this REST endpoint.
+	 *
+	 * @param string $check The endpoint check being ran.
+	 *
+	 * @return bool
+	 */
+	public function validate_request_permission( $check ) {
```

Comment:
```suggestion
	protected function validate_request_permission( $check ) {
```

We can make this a protected function.